arrow-site/docs/docs/patterns/errorhandling/README.md

```diff
@@ -104,70 +103,62 @@ Constructing an exception may be as costly as your current Thread stack size, an
 More info on the cost of instantiating Throwables, and throwing exceptions in general, can be found in the links below.
 
 > [The Hidden Performance costs of instantiating Throwables](http://normanmaurer.me/blog/2013/11/09/The-hidden-performance-costs-of-instantiating-Throwables/)
-> * New: Creating a new Throwable each time
-> * Lazy: Reusing a created Throwable in the method invocation.
-> * Static: Reusing a static Throwable with an empty stacktrace.
 
 Exceptions may be considered generally a poor choice in Functional Programming when:
 
 - Modeling absence
 - Modeling known business cases that result in alternate paths
-- Used in async boundaries over unprincipled APIs (callbacks)
-- In general, when people have no access to your source code
+- Used in async boundaries over APIs based callbacks that lack some form of structured concurrency.
+- In general, when people have no access to your source code.
 
 ### How do we model exceptional cases then?
 
-Arrow provides proper datatypes and typeclasses to represent exceptional cases.
+Arrow and the Kotlin standard library provides proper datatypes and abstractions to represent exceptional cases.
 
-### Option
+### Nullable types
 
-We use [`Option`]({{'/apidocs/arrow-core-data/arrow.core/-option/' | relative_url }}) to model the potential absence of a value.
+We use [`Nullable Types`](https://kotlinlang.org/docs/null-safety.html#nullable-types-and-non-null-types) to model the potential absence of a value.
 
-When using `Option`, our previous example may look like:
+When using `Nullable types`, our previous example may look like:
 
 ```kotlin:ank
-import arrow.*
-import arrow.core.*
-
-fun takeFoodFromRefrigerator(): Option<Lettuce> = None
-fun getKnife(): Option<Knife> = None
-fun prepare(tool: Knife, ingredient: Lettuce): Option<Salad> = Some(Salad)
+fun takeFoodFromRefrigerator(): Lettuce? = null
+fun getKnife(): Knife? = null
+fun prepare(tool: Knife, ingredient: Lettuce): Salad? = Salad
 ```
 
-It's easy to work with [`Option`]({{'/apidocs/arrow-core-data/arrow.core/-option/' | relative_url }}) if your lang supports [Monad Comprehensions]({{ '/patterns/monad_comprehensions' | relative_url }}) or special syntax for them.
-Arrow provides [monadic comprehensions]({{ '/patterns/monad_comprehensions' | relative_url }})  for all datatypes for which a [`Monad`]({{'/arrow/typeclasses/monad' | relative_url }}) instance exists built atop coroutines.
+It's easy to work with [`Nullable Types`](https://kotlinlang.org/docs/null-safety.html#nullable-types-and-non-null-types) if your lang supports special syntax like `?` as Kotlin does. 
+Nullable types are faster than boxed types like `Option`. Nonetheless `Option` is also supported by Arrow to interop with Java based libraries that use `null` as signal or interruption value like [ReactiveX RxJava](https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#nulls).
```

Comment:
Or to use in generic code when not constraining with `A : Any` and using `null` as a nested signal. Meaning `Option<Option<A>>` since `A?` can't exist.

Reply:
Added additional comment based on this info